File: rustv/src/front_end/ir.rs
use anyhow::Result;

use super::ast::{Expr, Opcode};

pub enum IR {
    OP_ADD,
    OP_SUB,
    OP_MUL,
    OP_DIV,
}

pub fn simulate(ast: Expr) -> Result<()> {
    match  ast {
        Expr::Number(_) => todo!(),
        Expr::Op(expr, opcode, expr1) => todo!(),
    }
}
pub fn binary(lhs: Box<Expr>, op: Opcode, rhs: Expr) -> Result<()> {
    let rhs_val = simulate(rhs);
    match op {
        Opcode::Add => todo!(),
        Opcode::Sub => todo!(),
        Opcode::Mul => todo!(),
        Opcode::Div => todo!(),
    }
}

File: gov/compiler/ir.go
package parser

type IRType string

const (
	OP_PUSH IRType = "OP_PUSH"
	OP_ADD  IRType = "OP_ADD"
	OP_SUB  IRType = "OP_SUB"
	OP_MUL  IRType = "OP_MUL"
	OP_DIV  IRType = "OP_DIV"
)
